java-squid/src/main/java/org/sonar/java/se/SymbolicValue.java

```diff
@@ -225,6 +225,21 @@ public String toString() {
     BooleanConstraint shouldNotInverse() {
       return BooleanConstraint.FALSE;
     }
+
+    @Override
+    public List<ProgramState> setConstraint(ProgramState programState, BooleanConstraint booleanConstraint) {
+      ProgramState ps = programState;
+      if (leftOp.id > 2 && rightOp.id > 2) {
```

Comment:
This is rather cryptic and really likely to be error prone (changing order of constants SV will screw up behaviour of engine). Please at least factorize this in a dedicated method and use the SV constants.